src/qat/purr/qatconfig.py

```diff
@@ -1,13 +1,36 @@
-from typing import Optional
+from typing import Literal, Optional
 
-from pydantic import Field, field_validator
+from pydantic import BaseModel, ConfigDict, Field, field_validator
 from pydantic_settings import BaseSettings, SettingsConfigDict
+from qiskit_aer import AerSimulator
 
 from qat.purr.utils.logger import get_default_logger
 
 log = get_default_logger()
 
 
+class QatSimulationConfig(BaseModel):
```

Comment:
Question: How generic and reusable this config specification is ?
I mean can I use this config maybe in the future to specify simulations against a backend other than AerSimulator?
If no then that means there needs to be semantic cushioning between this config and how AerSimulator was actually specified.
Might not be the case, qatconfig is part of the public API, we'd need to avoid annoying renames, config repurposes, which would only lead to deprecations and headaches.
If it's only specific for Qiskit, then please rename to `QiskitSimulationConfig`

Reply:
Good point. While the MPS configuration would be reusable, I certainly wouldn't want to call the arguments by their Qiskit names. Lots of the other parameters are specific to the many implementations AerSimulator can call. Let's go with `QiskitSimulationConfig`. Thanks!